GVFS/GVFS.Common/Maintenance/LooseObjectsStep.cs

```diff
@@ -0,0 +1,120 @@
+﻿using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.Common.Maintenance
+{
+    public class LooseObjectsStep : GitMaintenanceStep
+    {
+        public const string LooseObjects = "loose-objects.history";
+        private readonly string looseObjectsHistoryPath = null;
+        private readonly bool ignoreTimeRestriction = false;
+
+        public LooseObjectsStep(GVFSContext context, bool requireCacheLock, bool ignoreTimeRestriction)
+            : this(context, requireCacheLock)
+        {
+            this.ignoreTimeRestriction = ignoreTimeRestriction;
+        }
+
+        public LooseObjectsStep(GVFSContext context, bool requireCacheLock)
+            : base(context, gitObjects: null, requireObjectCacheLock: requireCacheLock)
+        {
+            this.looseObjectsHistoryPath = Path.Combine(this.Context.Enlistment.GitObjectsRoot, "info", LooseObjects);         
+        }
+
+        public override string Area => "LooseObjectsStep";
+
+        public int CountLooseObjects()
+        {
+            int count = 0;
+            int pathLength = this.Context.Enlistment.GitObjectsRoot.Length;
+            
+            foreach (string directory in this.Context.FileSystem.EnumerateDirectories(this.Context.Enlistment.GitObjectsRoot))
+            {
+                // Check if the directory is 2 letter HEX

```

Comment:
Instead of a regex, could we just iterate through the 256 options and check existence before enumerating?

Reply:
Thanks all!  I like @wilbaker idea of making a function in SHA1Util that can also be used DiagnoseVerb to validate the directory name and I've updated accordingly. 

Long term I think a shared LogLooseObjectCount function makes a ton of sense.  It would require some retweaking in DiagnoseVerb, so I'm saving that for a later iteration.  I think it's definitely worth doing.

I'll let you know when the changes are pushed up.